### ISSUE TITLE
Throw java.lang.NoSuchMethodError when run in Spring Boot

### ISSUE THREAD

**Issue by huantt** [jhalterman/failsafe#185]
When I create a Component in Spring project then run, it will throw `java.lang.NoSuchMethodError: net.jodah.failsafe.Failsafe.with([Ljava/lang/Object;)Lnet/jodah/failsafe/FailsafeExecutor;`

### Full logs:
```console
   java.lang.NoSuchMethodError: net.jodah.failsafe.Failsafe.with([Ljava/lang/Object;)Lnet/jodah/failsafe/FailsafeExecutor;
        at io.x10.pushtimize.webpushprocessor.processor.DemoRetry.run(DemoRetry.groovy:35) ~[main/:na]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_161]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_161]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_161]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_161]
        at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:84) ~[spring-context-5.1.5.RELEASE.jar:5.1.5.RELEASE]
        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) ~[spring-context-5.1.5.RELEASE.jar:5.1.5.RELEASE]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_161]
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [na:1.8.0_161]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [na:1.8.0_161]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [na:1.8.0_161]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_161]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_161]
        at java.lang.Thread.run(Thread.java:748) [na:1.8.0_161]

```

**Comment by huantt** [comment]
Note:
- I'm using Groovy
- I'm using @Compilestatic with all classes

**Comment by huantt** [comment]
I have found a workaround: Remove @Compilestatic

**Comment by huantt** [comment]
This is a problem of Groovy, that has been resolved with Groovy 2.5.6
https://issues.apache.org/jira/browse/GROOVY-7160
